src/main/java/com/splunk/ibm/mq/opentelemetry/Main.java

```diff
@@ -59,7 +63,15 @@ public static void main(String[] args) {
       initialDelaySeconds =
           YmlUtils.getInt(taskSchedule.get("initialDelaySeconds"), initialDelaySeconds);
     }
-    final ScheduledExecutorService service = Executors.newScheduledThreadPool(numberOfThreads);
+    Thread.UncaughtExceptionHandler handler = (t, e) -> logger.error(e.getMessage(), e);
```

Comment:
The message is already printed by the logger when you pass in the throwable, so let's make the message a bit more contextual and not redundant:

```suggestion
    Thread.UncaughtExceptionHandler handler = (t, e) -> logger.error("Unhandled exception in thread pool", e);
```